go/stats/prombackend/prombackend.go

```diff
@@ -0,0 +1,160 @@
+
+// NewMultiCounter is part of the PullBackend interface.
+func (be *PromBackend) NewMultiCounter(mc *stats.MultiCounters, name string) {
+	c := &multiCountersCollector{
+		multiCounters: map[*stats.MultiCounters]*prom.Desc{
+			mc: prom.NewDesc(
+				prom.BuildFQName("", be.namespace, toSnake(name)),
+				mc.Counters.Help(),
+				labelsToSnake(mc.Labels()),
+				nil),
+		}}
+
+	prom.MustRegister(c)
+}
+
+// NewMultiGauge is part of the PullBackend interface.
+func (be *PromBackend) NewMultiGauge(mg *stats.MultiGauges, name string) {
+	c := &multiGaugesCollector{
+		multiGauges: map[*stats.MultiGauges]*prom.Desc{
+			mg: prom.NewDesc(
+				prom.BuildFQName("", be.namespace, toSnake(name)),
+				mg.Gauges.Counters.Help(),
+				labelsToSnake(mg.Labels()),
+				nil),
+		}}
+
+	prom.MustRegister(c)
+}
+
+// NewMultiCounterFunc is part of the PullBackend interface.
+func (be *PromBackend) NewMultiCounterFunc(mcf *stats.MultiCountersFunc, name string) {
+	collector := &multiCountersFuncCollector{
+		multiCountersFunc: map[*stats.MultiCountersFunc]*prom.Desc{
+			mcf: prom.NewDesc(
+				prom.BuildFQName("", be.namespace, toSnake(name)),
+				mcf.Help(),
+				labelsToSnake(mcf.Labels()),
+				nil),
+		}}
+
+	prom.MustRegister(collector)
+}
+
+// NewTiming is part of the PullBackend interface
+func (be *PromBackend) NewTiming(t *stats.Timings, name string) {
+	collector := &timingsCollector{
+		timings: map[*stats.Timings]*prom.Desc{
+			t: prom.NewDesc(
+				prom.BuildFQName("", be.namespace, toSnake(name)),
+				t.Help(),
+				[]string{"Histograms"}, // hard coded label key
+				nil),
+		}}
+
+	prom.MustRegister(collector)
+}
+
+// NewMultiTiming is part of the PullBackend interface
+func (be *PromBackend) NewMultiTiming(mt *stats.MultiTimings, name string) {
+	collector := &multiTimingsCollector{
+		multiTimings: map[*stats.MultiTimings]*prom.Desc{
+			mt: prom.NewDesc(
+				prom.BuildFQName("", be.namespace, toSnake(name)),
+				mt.Help(),
+				labelsToSnake(mt.Labels()),
+				nil),
+		}}
+
+	prom.MustRegister(collector)
+}
+
+// NewInt is part of the PullBackend interface
+func (be *PromBackend) NewInt(i *stats.Int, name string, vt stats.ValueType) {
+	collector := &intCollector{
+		intMetrics: map[*stats.Int]*prom.Desc{
+			i: prom.NewDesc(
+				prom.BuildFQName("", be.namespace, toSnake(name)),
+				i.Help(),
+				nil,
+				nil),
+		}, vt: vt}
+	prom.MustRegister(collector)
+}
+
+// NewIntFunc is part of the PullBackend interface
+func (be *PromBackend) NewIntFunc(intFunc *stats.IntFunc, name string) {
+	collector := &intFuncCollector{
+		intFuncs: map[*stats.IntFunc]*prom.Desc{
+			intFunc: prom.NewDesc(
+				prom.BuildFQName("", be.namespace, toSnake(name)),
+				intFunc.Help(),
+				nil,
+				nil),
+		}}
+
+	prom.MustRegister(collector)
+}
+
+func labelsToSnake(labels []string) []string {
+	output := make([]string, len(labels))
+	for i, l := range labels {
+		output[i] = toSnake(l)
+	}
+	return output
+}
+
+func toSnake(name string) string {
```

Comment:
@michael-berlin is there a way you can export youtube's function that converts to snake-case? I'm assuming that it shouldn't have anything proprietary.

Reply:
Moving the snake case / kebab-case conversation here:

> (From @michael-berlin in Reviewable:) Turns out snake_case uses underscores but internally we're doing kebab-case (with hyphens).
I've exported the code: #3816
But I'm not sure if it's worth it merging it.
Also note that we do not have special handling for certain words e.g. the resulting internal variable name is vtgate-v-schema-counts :)

Ooh. Thanks for exporting! 

@sougou Thoughts here? I can also just change my `toSnakeCase` to follow what `toKebabCase` is doing with the `regexp` package (which looks significantly cleaner than my hack with runes), and add a test. 

I would also prefer to do some special casing (For things like VSchema, VtGate) as part of the Prometheus export.